fix(billing): Fix toggling usage/spend charts for reserved budgets

dashed opened the pull request:
<!-- Describe your PR here. -->

dashed committed:
fix(billing): Fix toggling usage/spend charts for reserved budgets

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes

dashed committed:
fix

isabellaenriquez commented on static/gsApp/views/subscriptionPage/reservedUsageChart.tsx:
```diff
@@ -463,9 +463,6 @@ function ReservedUsageChart({
   const categoryStats = usageStats[category];
   const isReservedBudgetCategory =
     subscription.reservedBudgetCategories?.includes(category) ?? false;
-  if (isReservedBudgetCategory) {
-    displayMode = 'cost';
-  }
```
should we keep the override for sales-led customers since they don't have the toggle option?

isabellaenriquez reviewed (commented):
lgtm overall, just a couple test questions and one ux question

dashed committed:
remove

dashed commented:
Thanks for the great feedback! I've addressed all three concerns:

## 🔧 Sales-led Customer Override (Line 468)

**Issue**: Should we keep the override for sales-led customers since they don't have the toggle option?

**Solution**: ✅ Added back the override logic for sales-led customers (`canSelfServe: false`) to force `displayMode = 'cost'` for reserved budget categories, since they don't have access to the usage/cost toggle in the UI.

```typescript
// For sales-led customers (canSelfServe: false), force cost view for reserved budget categories
// since they don't have access to the usage/cost toggle
if (isReservedBudgetCategory && !subscription.canSelfServe) {
  displayMode = 'cost';
}
```

## 🧪 Test Issues (Lines 770/853)

**Issue**: Why override what's set in `SubscriptionWithSeerFixture`? `reserved` should be `RESERVED_BUDGET_QUOTA`.

**Solution**: ✅ Updated tests to use fixture values instead of manual overrides:

- Created a helper function `getReservedBudgetCategoryInfo()` to extract values from the subscription's `reservedBudgets`
- Updated both SEER tests to use `getReservedBudgetCategoryInfo(subscription)` instead of manual `reservedBudgetCategoryInfo` objects
- Added a new test to verify sales-led customer behavior

This ensures tests properly use the `RESERVED_BUDGET_QUOTA` (-2) and other values set by `SubscriptionWithSeerFixture`.

## ✅ All Tests Passing

Added comprehensive test coverage for:

- Self-serve customers can toggle between usage/cost views ✅
- Sales-led customers are forced to cost view ✅
- Tests use fixture values instead of manual overrides ✅

The fix now properly:

- Allows self-serve customers to toggle between usage/cost views for reserved budgets
- Forces sales-led customers to cost view (since they can't toggle)
- Maintains backward compatibility and proper UX for different customer types

dashed committed:
force cost display mode for sales-led customers

dashed committed:
fix lint
